modules/rostests/apitests/ntdll/NtQueryInformationProcess.c

```diff
@@ -185,6 +185,50 @@ Test_ProcessTimes(void)
 #undef SPIN_TIME
 }
 
+static
+void
+Test_ProcessPriorityClassAlignment(void)
+{
+    NTSTATUS Status;
+    PPROCESS_PRIORITY_CLASS ProcPriority;
+
+    /* Allocate some memory for the priority class structure */
+    ProcPriority = malloc(sizeof(PROCESS_PRIORITY_CLASS));
+    if (ProcPriority == NULL)
+    {
+        skip("Failed to allocate memory for PROCESS_PRIORITY_CLASS!\n");
+        return;
+    }
+
+    /* Unaligned buffer -- wrong size */
+    Status = NtQueryInformationProcess(NtCurrentProcess(),
+                                       ProcessPriorityClass,
+                                       (PVOID)1,
+                                       0,
+                                       NULL);
+    ok_hex(Status, STATUS_INFO_LENGTH_MISMATCH);
+
+    /* Unaligned buffer -- correct size */
+    Status = NtQueryInformationProcess(NtCurrentProcess(),
+                                       ProcessPriorityClass,
+                                       (PVOID)1,
+                                       sizeof(PROCESS_PRIORITY_CLASS),
+                                       NULL);
+    ok_hex(Status, STATUS_DATATYPE_MISALIGNMENT);
+
+    /* Do not care for the length but expect to return the priority class */
+    Status = NtQueryInformationProcess(NtCurrentProcess(),
+                                       ProcessPriorityClass,
+                                       &ProcPriority,
```

Comment:
no `&` here.
Why it works perfectly in 2k3 / xp is because you don't run a debug version of those.